backend/core/src/main/java/org/sonarsource/sonarlint/core/remediation/aicodefix/AiCodeFixService.java

```diff
@@ -53,15 +56,18 @@ public class AiCodeFixService {
   private final PreviouslyRaisedFindingsRepository previouslyRaisedFindingsRepository;
   private final ClientFileSystemService clientFileSystemService;
   private final StorageService storageService;
+  private final TelemetryService telemetryService;
 
   public AiCodeFixService(ConnectionConfigurationRepository connectionRepository, ConfigurationRepository configurationRepository, ConnectionManager connectionManager,
-    PreviouslyRaisedFindingsRepository previouslyRaisedFindingsRepository, ClientFileSystemService clientFileSystemService, StorageService storageService) {
+    PreviouslyRaisedFindingsRepository previouslyRaisedFindingsRepository, ClientFileSystemService clientFileSystemService, StorageService storageService,
+    TelemetryService telemetryService) {
     this.connectionRepository = connectionRepository;
     this.configurationRepository = configurationRepository;
     this.connectionManager = connectionManager;
     this.previouslyRaisedFindingsRepository = previouslyRaisedFindingsRepository;
     this.clientFileSystemService = clientFileSystemService;
     this.storageService = storageService;
+    this.telemetryService = telemetryService;
```

Comment:
As telemetry is a separate concern, usually we rely on events to decouple the functionality and the associated telemetry